### PR TITLE
Use Prettier

### LINKED ISSUE #649: Switch to prettier for code formatting
[Prettier](https://github.com/prettier/prettier) has stabilized quite a bit and is now used by a few major projects and soon by nteract as well (https://github.com/nteract/nteract/issues/1552).
It can be set up with commit hooks, CI integration and also has a Atom plugin.

Let's switch to it!

### PR DESCRIPTION
**Pull request by lgeiger**
This adapts https://github.com/nteract/nteract/pull/1595 and https://github.com/nteract/nteract/pull/1598 for Hydrogen.

Thanks @rgbkrk @peggyrayzis for adding this to nteract 👍 🎉 

Closes #649

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,13 +33,21 @@
...
+    "precommit": "lint-staged"
   },
   "repository": "https://github.com/nteract/hydrogen",
   "license": "MIT",
   "engines": {
     "atom": ">=1.6.0 <2.0.0"
   },
+  "lint-staged": {
+    "*.js": [
+      "prettier --single-quote --write",
```
Comment (lgeiger) [inline package.json]: I used single quotes here to keep the diff smaller, though I'm happy to change that.

File: .eslintrc.yml [hunk 2]
```diff
@@ -20,12 +21,6 @@ settings:
...
-  no-underscore-dangle: [0]
-  no-useless-escape: [0]
-  prefer-rest-params: [0]
-  class-methods-use-this: [0]
-  react/jsx-filename-extension: [0]
-  jsx-a11y/no-static-element-interactions: [0]
-  react/no-danger: [0]
+  prettier/prettier:
+    - error
+    - singleQuote: true
```
Comment (rgbkrk) [inline .eslintrc.yml]: You sneaky.

File: package.json (hunk 1 above)
Comment (rgbkrk) [inline package.json]: Now I see. :smile:

**Comment by rgbkrk** [comment]
I use prettier atom with the defaults and love the no-fuss across projects. People would have to toggle their settings if they're using the prettier plugin, unless they have commit hooks set up as well.

**Comment by lgeiger** [comment]
Oh I see, I thought prettier-atom respects a per project config like eslint does.

Will change it to be consistent with nteract and other projects using prettier 👍

**Comment by rgbkrk** [comment]
Ah, didn't realize that.

**Pull request merged**